packages/sanity/src/core/bundles/hooks/usePerspective.tsx

```diff
@@ -15,29 +15,32 @@ export interface PerspectiveValue {
 }
 
 /**
+ * TODO: Improve distinction between global and pane perspectives.
+ *
  * @internal
  */
-export function usePerspective(): PerspectiveValue {
+export function usePerspective(selectedPerspective?: string): PerspectiveValue {
```

Comment:
Thank you! Adding this parameter was a bandaid, and I did plan to come back with _some_ more elegant solution. I think your suggestion is definitely much better.


I also wonder whether requiring the local override at all is a bit of a code smell 🤔.

Reply:
Yeah I'm not too happy with this solution either, I was trying to think of a way to remove this, I think it would require a refactor in the bundleMenu given is the only one that needs to work in two cases (local and global).

I was thinking on the option to add another hook in structure to handle only the "pane" perspective and maybe we could rename this one to `useGlobalPerspective` 

They will have a very similar behavior and could expose the same actions, but each one will handle the actions in a different way. 

Anyways, I think it could be a follow up and this shouldn't block your changes, seeing that change alone will be easier to identify the best approach.